### DISCUSSION TITLE
Panel UHD/FHD mode switch

### DISCUSSION THREAD

**Discussion by raraavis1337** [seerge/g-helper#2485] [Q&A] [answered]
Hi guys,
I have ASUS Zephyrus Duo 16 GX650RX, where the primary display supports UHD 120Hz and FHD 240Hz modes. In AC there was a button to switch between the two (over restart). 
Now I've just moved from AC to g-helper, and don't see this option anywhere... (Apologies in advance if I'm just missing it in front of my eyes)
P.S. If this functionality is not supported, is there another way to access this feature without having to reinstall AC? (command line maybe)
Thanks.

**Comment by seerge** [comment]
@raraavis1337 hello, currently app is not aware of that.

Can you switch resolution and refresh-rate of the second screen via windows settings ?

**Comment by raraavis1337** [comment]
Hi @seerge 
Yes I can, but please help me understand why the second screen might be related?
The problem is with the primary screen, the secondary screen pad is turned off.
To my understanding, that UHD/FHD switch is some kind of hardware configuration rather than a resolution change. So that if I just decrease the resolution from UHD to FHD without changing the mode, both Windows and g-helper still allow me choosing between 120 and 60Hz,  while the 240Hz does not become available.
// tbh I'm a bit surprised that no one is asking about such hybrid dual-mode panels, as it is quite common on premium non-OLED gaming laptops - i.e. I have it on my other Razer Blade 16 2024 mini led as well... 

I will be happy to provide any additional information or to test if you want to consider working on that functionality. 

Thanks!

**Reply by seerge** [reply]
@raraavis1337 if it's some unique hardware switch you can run endpoint discovery scan and upload DSDT dump as explained here 
https://github.com/seerge/g-helper/issues/38

**Reply by raraavis1337** [reply]
Ran both methods, results attached. Thanks.
[gx650rx-endpoints.zip](https://github.com/seerge/g-helper/files/15081803/gx650rx-endpoints.zip)

**Reply by seerge** [reply, marked as answer]
@raraavis1337 ok, thanks

Do I understand correctly that to actually change Resolution/Refresh rate AC required restart ?

Can you run following command in powershell as admin : 

```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DSTS -Arguments @{Device_ID=0x0005001C}
```

And check what it outputs ? it should return UHD / FHD switch status

Then to do actual switch run in powershell as admin :

```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DEVS -Arguments @{Device_ID=0x0005001C; Control_status=1}
```

and ``Restart``

Then run
```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DEVS -Arguments @{Device_ID=0x0005001C; Control_status=0}
```

and ``Restart`` again

And check if it actually changes screen status ?

``Control_status`` param is actual status of the switch :) it can be 1 or 0

**Comment by raraavis1337** [comment]
@seerge   Yeap, spot on, that's the flag! And it works as expected.
Also G-Helper reads the max refresh correctly in both modes.
(The only thing is that it give an option of either 60, or 240+OD, or auto for 60 on battery and 240 plugged-in. While the display supports 5 modes: fixed 60, 120, and 240, auto 60/120, and auto 120/240). You might consider adding all of them, or at least the 120Hz fixed.) 
Thanks a lot, you are amazing! :)

**Reply by seerge** [reply]
@raraavis1337 ok, good news.

Is 0 = UHD 120Hz and 1 = FHD 240HZ ?

As for refresh rate buttons, they are universal for all devices. It's always 60Hz and [Maximum available in windows]Hz only.

**Reply by seerge** [reply]
P.S. Do I understand that your device doesn't have a Mini-led display (and therefore you don't see a toggle for that) ? I can add 4th button in display row for togging between FHD / UHD

**Reply by raraavis1337** [reply]
> Is 0 = UHD 120Hz and 1 = FHD 240HZ ?

That's correct. 

> P.S. Do I understand that your device doesn't have a Mini-led display (and therefore you don't see a toggle for that) ?

Yes, that's also correct. It's an IPS display, panel [BOE NE160QAM-NX1](https://www.panelook.com/NE160QAM-NX1_BOE_16.0_LCM_overview_55903.html). Full spec of the laptop [here](https://www.notebookcheck.net/Asus-ROG-Zephyrus-Duo-16-review-An-unusual-gaming-laptop-with-a-good-battery-life.621662.0.html).

**Comment by seerge** [comment]
@raraavis1337 ok, try this build

It should show a button-toggle for UHD / FHD and do switch / reboot when you click

[GHelper.zip](https://github.com/seerge/g-helper/files/15083076/GHelper.zip)

**Comment by raraavis1337** [comment]
> It should show a button-toggle for UHD / FHD and do switch / reboot when you click

Tested, functionally it is working well.
From the UX perspective, a bit counter-intuitive. The button says "FHD" while in FHD mode, and "UHD" while in UHD mode. Probably better to display something like "to UHD" while in FHD mode, and "to FHD" in UHD mode + add some explanation in the tooltip line below.

**Reply by seerge** [reply]
@raraavis1337 button shows current state. From my point of view showing opposite may be confusing. 

Current behavior is similar to Multi-zone switch for miniled displays :) 

I have added a tooltip under button (similar to refresh rates tooltips) explaining what does button do

[GHelper.zip](https://github.com/seerge/g-helper/files/15083379/GHelper.zip)

**Reply by raraavis1337** [reply]
It's perfect. Thanks a lot again!

**Comment by flukejones** [comment]
This seems like a confusing thing for a WMI method to be doing.. What does the WMI method actually do when toggled here?

**Reply by seerge** [reply]
@flukejones  WMI method is needed to call same endpoint that you can access via ACPI directly.

Endpoint that you need is ``0x0005001C`` and it's an ON / OFF toggle for screen FHD "mode"
